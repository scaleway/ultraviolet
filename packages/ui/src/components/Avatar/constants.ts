import type { Theme } from '@emotion/react'

// Match the text variant with component size
export const TEXT_VARIANT_BY_SIZE = {
  large: 'headingLarge',
  medium: 'headingSmall',
  small: 'bodySmall',
  xsmall: 'captionSmall',
} as const

// Match the container size with actual px size
export const sizes = (theme: Theme) =>
  ({
    large: '7rem', // TODO: add this value to tokens
    medium: theme.sizing['800'],
    small: theme.sizing['400'],
    xsmall: theme.sizing['250'],
  }) as const

export const RADIUS_SIZES = {
  large: 'xxlarge',
  medium: 'xlarge',
  small: 'large',
  xsmall: 'default',
} as const

// Defines all available sentiments for the eligible variants
export const SENTIMENTS = ['primary', 'neutral'] as const

// It's the default color set for the colors variant
export const DEFAULT_COLORS = ['primary', 'secondary'] as const
