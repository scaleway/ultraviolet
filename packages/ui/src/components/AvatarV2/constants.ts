import type { Theme } from '@emotion/react'

// Match the text variant with component size
export const TEXT_VARIANT_BY_SIZE = {
  xsmall: 'captionSmall',
  small: 'bodySmall',
  medium: 'headingSmall',
  large: 'headingLarge',
} as const

// Match the container size with actual px size
export const sizes = (theme: Theme) =>
  ({
    xsmall: theme.sizing['250'],
    small: theme.sizing['400'],
    medium: theme.sizing['800'],
    large: '7rem', // TODO: add this value to tokens
  }) as const

export const RADIUS_SIZES = {
  xsmall: 'default',
  small: 'large',
  medium: 'xlarge',
  large: 'xxlarge',
} as const

// Defines all available sentiments for the eligible variants
export const SENTIMENTS = ['primary', 'neutral'] as const

// It's the default color set for the colors variant
export const DEFAULT_COLORS = ['primary', 'secondary'] as const
