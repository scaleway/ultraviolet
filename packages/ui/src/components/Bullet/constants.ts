// Values are then used as theme.sizing[X]
export const SIZES = {
  medium: '400',
  small: '300',
  xsmall: '250',
  xxsmall: '200',
} as const

export const PROMINENCES = {
  default: '',
  strong: 'Strong',
} as const
