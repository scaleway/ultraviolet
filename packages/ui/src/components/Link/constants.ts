export const PROMINENCES = {
  default: '',
  strong: 'strong',
  stronger: 'stronger',
  weak: 'weak',
} as const
