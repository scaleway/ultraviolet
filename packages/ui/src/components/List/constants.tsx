'use client'

export const SELECTABLE_CHECKBOX_SIZE = '500' // sizing token from theme
export const EXPANDABLE_COLUMN_SIZE = '500'
