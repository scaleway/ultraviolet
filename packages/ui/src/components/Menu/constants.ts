export const SIZES = {
  small: '11.25rem',
  medium: '17.5rem',
  large: '23.75rem',
}
