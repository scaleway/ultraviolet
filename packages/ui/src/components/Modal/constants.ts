import type { ModalSize } from './types'

export const MODAL_WIDTH: Record<ModalSize, number> = {
  large: 53.125, // in rem
  medium: 44.25,
  small: 38.5,
  xsmall: 25,
  xxsmall: 22.5,
}

export const MODAL_PLACEMENT = {
  bottom: {
    margin: 'auto',
    marginBottom: 0,
  },
  'bottom-left': {
    margin: 'auto',
    marginBottom: 0,
    marginLeft: 0,
  },
  'bottom-right': {
    margin: 'auto',
    marginBottom: 0,
    marginRight: 0,
  },
  center: {
    margin: 'auto',
  },
  left: {
    margin: 'auto',
    marginLeft: 0,
  },
  right: {
    margin: 'auto',
    marginRight: 0,
  },
  top: {
    margin: 'auto',
    marginTop: 0,
  },
  'top-left': {
    margin: 'auto',
    marginLeft: 0,
    marginTop: 0,
  },
  'top-right': {
    margin: 'auto',
    marginRight: 0,
    marginTop: 0,
  },
}
