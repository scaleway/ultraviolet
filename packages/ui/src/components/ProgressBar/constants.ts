export const PROGRESS_BAR_SENTIMENTS = [
  'primary',
  'success',
  'warning',
  'info',
  'danger',
] as const
