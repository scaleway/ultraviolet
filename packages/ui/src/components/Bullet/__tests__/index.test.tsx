import { MoonIcon, MoonOutlineIcon } from '@ultraviolet/icons'
import { shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, test } from 'vitest'
import { Bullet } from '..'
import { BULLET_SENTIMENTS } from '../constants'

describe('bullet', () => {
  test('renders correctly with a text', () =>
    shouldMatchEmotionSnapshot(<Bullet>1</Bullet>))

  test('renders correctly with an icon', () =>
    shouldMatchEmotionSnapshot(
      <Bullet>
        <MoonIcon />
      </Bullet>,
    ))

  test('renders correctly with an icon and icon variant', () =>
    shouldMatchEmotionSnapshot(
      <Bullet>
        <MoonOutlineIcon />
      </Bullet>,
    ))

  describe('sentiment', () => {
    BULLET_SENTIMENTS.forEach(sentiment => {
      test(`render ${sentiment}`, () =>
        shouldMatchEmotionSnapshot(<Bullet sentiment={sentiment}>1</Bullet>))
    })
  })

  describe('size', () => {
    ;(['medium', 'small', 'xsmall', 'xxsmall'] as const).forEach(size => {
      test(`render ${size}`, () =>
        shouldMatchEmotionSnapshot(<Bullet size={size}>1</Bullet>))
    })
  })
})
