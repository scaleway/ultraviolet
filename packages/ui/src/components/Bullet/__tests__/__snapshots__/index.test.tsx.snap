// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bullet > renders correctly disabled 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_disabled_true__1jy42m0e styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > renders correctly with a text 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_2__1jy42m0h"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > renders correctly with an icon 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_2__1jy42m0h"
    >
      <svg
        class="styles__vbm0wq0 styles_prominence_default__vbm0wqg styles_disabled_false__vbm0wqf styles_size_small__vbm0wqa styles_undefined_compound_39__vbm0wq1n"
        viewBox="0 0 16 16"
      >
        <path
          d="M14.9543 10.6781C15.1672 10.2014 14.5937 9.79573 14.1149 10.0038C13.3621 10.3309 12.5313 10.5123 11.6581 10.5123C8.25024 10.5123 5.48762 7.74964 5.48762 4.34178C5.48762 3.46865 5.66897 2.63787 5.99605 1.88507C6.20408 1.40628 5.79836 0.832829 5.32171 1.0457C2.77491 2.1831 1 4.73805 1 7.70756C1 11.735 4.26491 14.9999 8.29238 14.9999C11.2619 14.9999 13.8169 13.225 14.9543 10.6781Z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > renders correctly with an icon and icon variant 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_2__1jy42m0h"
    >
      <svg
        class="styles__vbm0wq0 styles_prominence_default__vbm0wqg styles_disabled_false__vbm0wqf styles_size_small__vbm0wqa styles_undefined_compound_39__vbm0wq1n"
        viewBox="0 0 16 16"
      >
        <path
          clip-rule="evenodd"
          d="M6.13247 1.17958C6.30822 1.35534 6.36047 1.61981 6.26475 1.84919C5.96713 2.56244 5.80246 3.34567 5.80246 4.16892C5.80246 7.49844 8.50157 10.1975 11.8311 10.1975C12.6543 10.1975 13.4376 10.0329 14.1508 9.73525C14.3802 9.63953 14.6447 9.69178 14.8204 9.86753C14.9962 10.0433 15.0484 10.3078 14.9527 10.5371C13.8595 13.1569 11.2731 15 8.25478 15C4.24808 15 1 11.7519 1 7.74522C1 4.72688 2.84311 2.14046 5.46286 1.0473C5.69224 0.951585 5.95672 1.00383 6.13247 1.17958ZM4.69006 2.88277C3.19547 3.98037 2.22616 5.75005 2.22616 7.74522C2.22616 11.0747 4.92526 13.7738 8.25478 13.7738C10.25 13.7738 12.0196 12.8045 13.1172 11.3099C12.6996 11.3847 12.2697 11.4237 11.8311 11.4237C7.82438 11.4237 4.5763 8.17563 4.5763 4.16892C4.5763 3.73027 4.6153 3.30043 4.69006 2.88277Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > sentiment > render danger 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_4__1jy42m0j"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > sentiment > render info 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_6__1jy42m0l"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > sentiment > render neutral 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_2__1jy42m0h"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > sentiment > render primary 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_0__1jy42m0f"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > sentiment > render secondary 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_1__1jy42m0g"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > sentiment > render success 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_3__1jy42m0i"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > sentiment > render warning 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_5__1jy42m0k"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > size > render medium 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_medium__1jy42m01 styles_prominence_default__1jy42m0c styles_undefined_compound_2__1jy42m0h"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > size > render small 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_small__1jy42m02 styles_prominence_default__1jy42m0c styles_undefined_compound_2__1jy42m0h"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > size > render xsmall 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_xsmall__1jy42m03 styles_prominence_default__1jy42m0c styles_undefined_compound_2__1jy42m0h"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;

exports[`bullet > size > render xxsmall 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__1jy42m00 styles_size_xxsmall__1jy42m04 styles_prominence_default__1jy42m0c styles_undefined_compound_2__1jy42m0h"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;
